src/main/kotlin/org/jetbrains/kotlinx/jupyter/apiImpl.kt

```diff
@@ -132,6 +136,23 @@ class NotebookImpl(
     override val jreInfo: JREInfoProvider
         get() = JavaRuntime
 
+    fun updateVarsState(varsMap: Map<String, VariableState>) {
```

Comment:
`varsMap` and `variablesMap` are semantically same things, but in this context they mean different entities. `varsMap` is a "state update", but `variablesMap` is a "whole state". Please rethink the naming you're using. Naming should be a reflection of variable meaning and of the reason it was created, not just a way of referencing

Reply:
Yeah, seems like I have a struggle not to think of better naming once a task is done. Algorithms contents flashbacks be like